mlflow/transformers/__init__.py

```diff
@@ -678,11 +748,15 @@ def save_model(
     if conda_env is None:
         if pip_requirements is None:
             default_reqs = get_default_pip_requirements(built_pipeline.model)
-            # NB: Skip inferring requirements for PEFT models to avoid loading the full pretrained
-            # model into memory. PEFT is mainly designed for fine-tuning large models under limited
-            # computational resources, so loading the full model is not preferred and can even crash
-            # the process due to OOM.
-            if not is_peft_model(built_pipeline.model):
+            if isinstance(transformers_model, str) or is_peft_model(built_pipeline.model):
+                _logger.info(
+                    "A local checkpoint path or PEFT model is given as the `transformers_model`. "
```

Comment:
Add to docs as well. There are a lot of caveats to this that remove many of the protections that were put into place based on early user confusion.

Reply:
Right, I will note that to the doc.